public/app/features/query/components/QueryGroup.tsx

```diff
@@ -241,7 +241,7 @@ export class QueryGroup extends PureComponent<Props, State> {
 
   onAddQuery = (query: Partial<DataQuery>) => {
     const { dsSettings, queries } = this.state;
-    this.onQueriesChange(addQuery(queries, query, { type: dsSettings?.type, uid: dsSettings?.uid }));
+    this.onQueriesChange(addQuery(queries, query, dsSettings ? getDataSourceRef(dsSettings) : {}));
```

Comment:
I actually believe the correct usage here would be to do `dsSettings ? getDataSourceRef(dsSettings) : undefined` since `addQuery` checks only if the object equals to [something truthy to set it](https://github.com/grafana/grafana/blob/apiVersionDatasourceRef/public/app/core/utils/query.ts#L27). In general, it clutters the code to maintain the `undefined` behavior (not only here but in the rest of places in which something similar happens) but well, we can keep it to be extra safe.